docs/usage/with-typescript.md

```diff
@@ -0,0 +1,224 @@
+---
+id: with-typescript
+title: Usage With TypeScript
+sidebar_label: Usage With TypeScript
+hide_title: true
+---
+
+# Usage With TypeScript
+
+It is one of Redux Toolkit's design targets to offer great integration with TypeScript.
+
+This page is intended to give an overview of all common usecases and the most probable pitfalls you might encounter when using RTK with TypeScript.
+
+**If you encounter any problems with the Types that are not described on this page, please open an Issue!**
+
+## Using configureStore with TypeScript
```

Comment:
Use inline code snippets for all function names in the headers, like:

> ## Using `configureStore` with TypeScript

Reply:
I believe I got all of those :+1: